openstack/image/v2/images/results.go

```diff
@@ -123,6 +125,20 @@ func (r *Image) UnmarshalJSON(b []byte) error {
 		return fmt.Errorf("Unknown type for SizeBytes: %v (value: %v)", reflect.TypeOf(t), t)
 	}
 
+	switch t := s.Hidden.(type) {
+	case nil:
+		r.Hidden = false
+	case bool:
+		r.Hidden = t
+	case string:
+		r.Hidden, err = strconv.ParseBool(t)
+		if err != nil {
+			return fmt.Errorf("Failed to parse Hidden %q: %v", t, err)
+		}
+	default:
+		return fmt.Errorf("Unknown type for Hidden: %v (value: %v)", reflect.TypeOf(t), t)
+	}
+
```

Comment:
~How is this happening. This should always be a boolean since that's how it's stored in the DB ([source](https://github.com/openstack/glance/blob/29.0.0/glance/db/sqlalchemy/models.py#L145-L146)) and the API is simply using exactly what is stored there ([source](https://github.com/openstack/glance/blob/29.0.0/glance/db/__init__.py#L96-L143)).~

Sorry, you explained this [in the issue](https://github.com/gophercloud/gophercloud/issues/3147#issuecomment-2252675540). It's been a while :sweat_smile: Would it be possible to get a comment here explaining what's going on for future us (without needing to search though GitHub Issues)